pkg/block/metadata/meta.go

```diff
@@ -62,25 +64,41 @@ type ThanosDownsample struct {
 	Resolution int64 `json:"resolution"`
 }
 
+// dirSize walks through the specified path recursively and returns
+// how many bytes does it take.
+func dirSize(path string) (uint64, error) {
+	var size uint64
+	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			size += uint64(info.Size())
+		}
+		return err
+	})
+	return size, err
+}
+
 // InjectThanos sets Thanos meta to the block meta JSON and saves it to the disk.
 // NOTE: It should be used after writing any block by any Thanos component, otherwise we will miss crucial metadata.
-func InjectThanos(logger log.Logger, bdir string, meta Thanos, downsampledMeta *tsdb.BlockMeta) (*Meta, error) {
-	newMeta, err := Read(bdir)
+func InjectThanos(logger log.Logger, bdir string, meta *Meta, downsampledMeta *tsdb.BlockMeta) (*Meta, error) {
```

Comment:
I think we can simplify this function signature.

I think we would be fine with just `Write` that would add `index size` and `chunks total size` right?

Reasons:
* We never now use `downsampledMeta` with your change as we reuse `meta` (which is nice)
* It's actually the same as Write now, except size, and actually it's nice and expected behaviour.

Reply:
Good catch! In fact, we had never passed the last argument to this function so I just went ahead and removed it, and renamed the existing users to use `Write()`.